workspaces/src/rpc/client.rs

```diff
@@ -37,11 +38,84 @@ impl Client {
         Self { rpc_addr }
     }
 
-    pub(crate) async fn query<M: methods::RpcMethod>(
+    async fn query_retry<M: methods::RpcMethod, F, T>(
         &self,
-        method: &M,
-    ) -> MethodCallResult<M::Response, M::Error> {
-        retry(|| async { JsonRpcClient::connect(&self.rpc_addr).call(method).await }).await
+        f: F,
+    ) -> MethodCallResult<M::Response, M::Error>
+    where
+        F: Fn() -> T,
+        T: core::future::Future<Output = MethodCallResult<M::Response, M::Error>>,
+    {
+        retry(|| async { f().await }).await
+    }
+
+    pub(crate) async fn query_broadcast_tx(
+        &self,
+        method: &methods::broadcast_tx_commit::RpcBroadcastTxCommitRequest,
+    ) -> MethodCallResult<
+        FinalExecutionOutcomeView,
+        near_jsonrpc_primitives::types::transactions::RpcTransactionError,
+    > {
+        self.query_retry::<methods::broadcast_tx_commit::RpcBroadcastTxCommitRequest, _, _>(|| async {
```

Comment:
since `retry(|| async { f().await }).await` can just be `retry(f).await`, which is like a thin wrapper that doesn't do much, we can just get rid of `query_retry` and do:
```suggestion
        retry(|| async {
```

Reply:
Yeah, good point, this common function is pretty useless